packages/flutter_tools/lib/src/hot.dart

```diff
@@ -43,6 +44,24 @@ class HotRunner extends ResidentRunner {
   ApplicationPackage _package;
   String _mainPath;
   final AssetBundle bundle = new AssetBundle();
+  final File pipe;
+
+  Future<String> _readFromControlPipe() async {
+    Stream<List<int>> stream = pipe.openRead();
+    List<int> bytes = await stream.first;
+    return new String.fromCharCodes(bytes);
+  }
+
+  Future<Null> _startReadingFromControlPipe() async {
+    if (pipe == null) {
+      return;
+    }
+    while (true) {
```

Comment:
add a comment saying how this loop ends (does _readFromControlPipe eventually throw? or do we rely on calling "exit" to end programs that use this?)